### DISCUSSION TITLE
Problems with pybricks block coding control two motor via a color sensor

### DISCUSSION THREAD

**Discussion by johnnyf64** [pybricks/support#1714] [Q&A]
Hello. I'm asking for some help with this. My setup is a train with two hubs. One for the two motors and a hub for the sensor. The sensor must determine when the train will move backwards when the yellow tile is reached. until he reaches the green tile and then again back to the yellow tile. With this setup he stops on the yellow tile and as soon as he loses the yellow tile the train wants to move forward again. He never reaches the green tile
![trainsetup](https://github.com/user-attachments/assets/ae45c189-6221-4c47-a3fc-ca612a164e24)

**Comment by laurensvalk** [comment]
> With this setup he stops on the yellow tile and as soon as he loses the yellow tile the train wants to move forward again. He never reaches the green tile

When the sensor is configured to detect only red/yellow/green, it is only going to report red/yellow/green. Even when it hovers over some track without a tile at all.

So it looks like the background colors are missing from your setup. Here is an example [from the tutorial](https://www.youtube.com/watch?v=kwnSbnlhOkA):

![image](https://github.com/user-attachments/assets/0505f05f-3f8b-4856-b4bb-82e8dfb9bf46)

It also might be a bit easier to test this with a single hub with a motor and a sensor first, just to be sure. And then expand to two hubs as you have.

**Reply by johnnyf64** [reply]
Is a background color necessary? I have a setup like you describe in the movie but without a background and it works perfectly. Is the block code I am using correct? The train needs more power so I decided to install a second motor for passing a switch and uncoupling a wagon

**Reply by laurensvalk** [reply]
You don't need add any physical bricks for background, but you'll just need to add any color that the train might possibly see in its path. In my case that's the tracks and the floor. It might be something else for you.

This is so it can distinguish those from the tiles. You could think of the tracks as another big tile.

Otherwise I think the sending/receiving part looks good.

**Reply by johnnyf64** [reply]
Hello, I cannot get the setup like above with a background to work. The train continues to go up and down within 1 sec. If I use a setup with 1 motor and one hub as Laurens describes in his YouTube video, it works fine. If I use my setup with one motor on hub1 and the sensor on hub2, the train does exactly the same. I don't think the block code works like that. Or I'm overlooking something

**Reply by johnnyf64** [reply]
Swapped the hubs and re-entered the code, and it finally works the way I want. I don't know what the problem was, but it works.

**Reply by laurensvalk** [reply]
Glad it worked!
